### PR TITLE
Hide all class logic

### PR DESCRIPTION
**Pull request by obrienu**
Hide all class logic. Refactor code to allow for jagged array by making colLength and rowLength dynamic.
Kindly review @bondz

### CODE REVIEW

File: src/rotten-tomatoes/SpoilTomatoes.js [hunk 1]
```diff
@@ -2,41 +2,40 @@ class SpoilTomatoes {
 
     constructor(grid) {
         this.grid = grid;
-        this.colLength = 0;
-        this.rowLength = 0;
         this.getAdjacents = this.getAdjacents.bind(this)
+        this.getRottenTomatoes = this.getRottenTomatoes.bind(this)
```
Comment (bondz) [inline src/rotten-tomatoes/SpoilTomatoes.js]: Didn't catch this before. Why do you need to do this? Its unnecessary in this context.

File: src/rotten-tomatoes/SpoilTomatoes.js [hunk 2]
```diff
@@ -2,41 +2,40 @@ class SpoilTomatoes {
...
     };
 
     getRottenTomatoes() {
-        const locationOfSpoiltTomatoes = [];
+        const spoiltTomatoesLocations = [];
         let numGoodTomatoe = 0;
         for (let row = 0; row < this.grid.length; row++) {
-            this.rowLength = this.grid.length - 1;
             for (let col = 0; col < this.grid[row].length; col++) {
```
Comment (bondz) [inline src/rotten-tomatoes/SpoilTomatoes.js]: With the assumption that the rowSize and columnSize is fixed for this grid, computing this every iterations can be avoided.

File: src/rotten-tomatoes/SpoilTomatoes.js [hunk 3]
```diff
@@ -55,7 +54,8 @@ class SpoilTomatoes {
                 let badTomatoe = arrOfBadTomatoes[i];
                 if (!visited[badTomatoe]) {
                     visited[badTomatoe] = true;
-                    const adjacentTomatoes = this.getAdjacents(badTomatoe);
+                    let [row,] = badTomatoe;
```
Comment (bondz) [inline src/rotten-tomatoes/SpoilTomatoes.js]: The comma is unnecessary

File: src/rotten-tomatoes/SpoilTomatoes.js [hunk 4]
```diff
@@ -55,7 +54,8 @@ class SpoilTomatoes {
...
+                    const adjacentTomatoes = this.getAdjacents(badTomatoe, this.grid[row].length - 1, this.grid.length - 1);
```
Comment (bondz) [inline src/rotten-tomatoes/SpoilTomatoes.js]: You can update the rowSize and columnSize in the constructor and just use it everywhere.

**Review by bondz — CHANGES_REQUESTED** [review]

File: src/rotten-tomatoes/SpoilTomatoes.js (hunk 2 above)
Comment (obrienu) [reply src/rotten-tomatoes/SpoilTomatoes.js]: Okay, I wrote the code this way with a consideration to a senerio where a jagged array way given. i'll make the changes

File: src/rotten-tomatoes/SpoilTomatoes.js (hunk 1 above)
Comment (obrienu) [reply src/rotten-tomatoes/SpoilTomatoes.js]: A habit i picked up from using class in React

**Comment by obrienu** [comment]
changes have been effected as requested @bondz . Kindly review

File: src/rotten-tomatoes/SpoilTomatoes.js [hunk 5]
```diff
@@ -2,41 +2,40 @@ class SpoilTomatoes {
...
+        this.rowLength = this.grid.length - 1;
     };
 
     getRottenTomatoes() {
-        const locationOfSpoiltTomatoes = [];
+        const spoiltTomatoesLocations = [];
         let numGoodTomatoe = 0;
         for (let row = 0; row < this.grid.length; row++) {
-            this.rowLength = this.grid.length - 1;
             for (let col = 0; col < this.grid[row].length; col++) {
```
Comment (bondz) [inline src/rotten-tomatoes/SpoilTomatoes.js]: `this.rowLength`, `this.colLength` can be used here now.

**Review by bondz — APPROVED** [review]

File: src/rotten-tomatoes/SpoilTomatoes.js (hunk 5 above)
Comment (obrienu) [reply src/rotten-tomatoes/SpoilTomatoes.js]: I already did that in the constructor, wont it be a repetition of code if i still reassign the same value here.

Comment (bondz) [reply src/rotten-tomatoes/SpoilTomatoes.js]: No assign, read.

Comment (obrienu) [reply src/rotten-tomatoes/SpoilTomatoes.js]: Okay, I have implemented the changes

**Pull request merged**